tests/core/checks/custom/BUILD.bazel

```diff
@@ -0,0 +1,150 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//tests:bazel_tests.bzl", "bazel_test")
+
+_workspace = """
+load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains")
+
+go_rules_dependencies()
+go_register_toolchains(go_checker="@//:my_checker")
+"""
+
+_single_check_build = """
+load("@io_bazel_rules_go//go:def.bzl", "go_checker", "go_tool_library")
+
+go_checker(
+    name = "my_checker",
+    deps = [":foo_func_check"],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "foo_func_check",
+    srcs = ["foo_func_check.go"],
+    importpath = "foofunccheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+"""
+
+_multiple_checks_build = """
+load("@io_bazel_rules_go//go:def.bzl", "go_checker", "go_tool_library")
+
+go_checker(
+    name = "my_checker",
+    deps = [
+        ":foo_func_check",
+        ":return_bool_check",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "foo_func_check",
+    srcs = ["foo_func_check.go"],
+    importpath = "foofunccheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "return_bool_check",
+    srcs = ["return_bool_check.go"],
+    importpath = "returnboolcheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+"""
+
+_no_error_check = """
+if [[ result -eq 1 ]]; then
+  echo "error: unexpected build error" >&2
+  result=1
+else
+  result=0
+fi
+"""
+
+_has_err_check_tmpl = """
+if [[ result -eq 0 ]]; then
+  echo "error: expected build error" >&2
+  result=1
+else
+  {check_err}
+fi
+"""
+
+_check_err_tmpl = """
+  grep -q '{err}' bazel-output.txt
+  if [[ $? -ne 0 ]]; then
+    echo "error: expected error message: '{err}'" >&2
+    result=1
+  else
+    result=0
+  fi
+"""
+
+
+bazel_test(
+    name = "single_check_has_errors",
+    build = _single_check_build,
+    workspace = _workspace,
+    check = _has_err_check_tmpl.format(
+        check_err =
+            _check_err_tmpl.format(err = "has_errors.go:4:1: function must not be named Foo")
+    ),
+    command = "build",
+    targets = [":has_errors"],
+    extra_files = [":foo_func_check.go"],
+)
+
+bazel_test(
+    name = "single_checks_no_errors",
+    build = _single_check_build,
+    workspace = _workspace,
+    check = _no_error_check,
+    command = "build",
+    targets = [":no_errors"],
+    extra_files = [":foo_func_check.go"],
+)
+
+bazel_test(
```

Comment:
Do the single check tests give much more coverage than the multiple check tests? `bazel_tests` are expensive enough to run that I think we should just have the two multiple-check cases.

I'll create a TODO for myself (#1508) to allow multiple tests in the same workspace. Then we could consolidate the `has_errors` and `no_errors` cases into a single test.

Reply:
Done. I thought that having both single and multiple checks would help us inductively reason about the behavior of the checking framework, but based on my knowledge of the implementation, running only the latter is probably fine.